prometheus/common/alert-rules.d/crunchy-alert-rules-etcd.yml.example

```diff
@@ -10,11 +10,58 @@ groups:
 
 ########## ETCD EXPORTER RULES ##########
 
+#### IMPORTANT NOTE: These alerts only support monitoring a single etcd cluster per prometheus instance ####
+
+# Ensure there is at least one etcd node reporting metrics
+  - alert: ETCDAbsent
+    expr: absent(etcd_server_has_leader)
+    for: 10s
+    labels:
+      service: etcd
+      severity: critical
+      severity_num: 300
+    annotations:
+      description: 'No etcd metric endpoints found'
+
+# Ensure there are an odd number of etcd nodes providing metrics
+  - alert: ETCDOddCount
+    expr: (count(etcd_server_has_leader) % 2) != 1
+    for: 10s
+    labels:
+      service: etcd
+      severity: critical
+      severity_num: 300
+    annotations:
+      description: 'An odd number of etcd nodes has not been found.'
+
+# Ensure there is a leader in the etcd cluster
+  - alert: ETCDLeader
+    expr: max(etcd_server_has_leader) < 1
+    for: 10s
+    labels:
+      service: etcd
+      severity: critical
+      severity_num: 300
+    annotations:
+      description: 'An etcd leader node has not been elected'
+
+# Ensure an expected amount of etcd nodes are running and providing metrics. Set this threshold to the amount of nodes in your cluster
+#
+#  - alert: ETCDMinimumNodeCount
```

Comment:
Hmm, this is named "Minimum Node Count", but really is total node count (at least from the comment). Seems like either the name or the comment ought to be adjusted accordingly, no?

Reply:
The actual value will be the total node count, yes, but the intention is to watch for a missing host in the cluster. The user must set the **intended** count of hosts in the cluster. Then when it goes below that, the alert fires. That's why this one was left commented out since it requires user interaction to set properly depending on the host count.